grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy

```diff
@@ -147,4 +154,21 @@ class DataUseLetterController implements ExceptionHandler, UserInfo {
         render(text: matches as JSON, contentType: "application/json")
     }
 
+
+    def findByUUID() {
+        DataUseLetter dul = DataUseLetter.findByUid(params.uuid)
+        Map<String, String> consent = new HashMap<>()
+        if(dul != null) {
+            User user = userService.findUser(dul.getCreator())
+            Issue issue = queryService.findByKey(dul.getConsentGroupKey())
+            consent.put("dataManagerName", user?.getDisplayName())
+            consent.put("dataManagerEmail", user?.getEmailAddress())
+            consent.put("consentGroupKey", issue?.getProjectKey())
+            consent.put(IssueExtraProperty.SUMMARY, issue?.getSummary())
+            consent.put(IssueExtraProperty.PROTOCOL, IssueExtraProperty.findByProjectKeyAndName(dul.getConsentGroupKey(), IssueExtraProperty.PROTOCOL).getValue())
+        }
+        response.status = 200
+        render([consent: consent] as JSON)
```

Comment:
Few details to point out in this method.
1. We shouldn't be using GORM's `IssueExtraProperty.findByProjectKeyAndName` here, better to use raw sql.
2. All of this data can be easily queried via raw sql.
3. `consent` doesn't seem like the right name here. We're returning an ad-hoc collection of user, project, and consent information. Given that this looks like a useful construct, it should be a full model object that can be returned by `QueryService`
4. Nitpicking: spacing around the `if` condition.